### PR TITLE
Remove specific file restriction

### LINKED ISSUE #196: Remove Specific File Extension Restriction
Specific file restriction logic currently exists within Frontary and needs to be removed to ensure it remains a general-purpose library, free from business-specific logic.

### PR DESCRIPTION
**Pull request by jinjoolee07**
Closes #196

### CODE REVIEW

File: src/input/user_input.rs [hunk 1]
```diff
@@ -807,6 +808,10 @@ where
             }
             Message::ChooseFile(my_index_clone.clone(), result, input_data_clone.clone())
         });
+        let extensions = extensions
+            .as_ref()
+            .unwrap_or(&vec![String::new()])
+            .join(",");
```
Comment (sophie-cluml) [inline src/input/user_input.rs:814]: I understand current logic as follows:

- If `extensions` is `None`, no extensions are allowed.
- If `extensions` is `Some`, only the extensions listed in the `Vec<String>` are allowed. This means that if the vector is empty, no extensions are allowed.

There is no functional difference between `None` and `Some(vec![])`. This suggests that the use of `Option` is unnecessary. 

Therefore, I suggest we either change `extensions` and `allowed_extensions` to just `Vec<String>`, or keep it as `Option<Vec<String>>` but change the behavior like below:

- If `extensions` is `None`, all extensions are allowed.

I believe one of these two directions would be more appropriate.

Comment (jinjoolee07) [reply src/input/user_input.rs]: Thank you for the feedback! I’ve implemented Option 1 by replacing `Option<Vec<String>>` with `Vec<String>`, simplifying the logic. Let me know if there’s anything else you’d like me to adjust!

File: src/input/user_input.rs [hunk 2]
```diff
@@ -830,8 +832,7 @@ where
                     <label for="input-file" class="input-file-choose-file">
                         { text!(txt, ctx.props().language, "Choose a file") }
                     </label>
-
-                    <input type="file" id="input-file" accept=".aice" name="upload-file" style="display: none;"
+                    <input type="file" id="input-file" accept={extensions.clone()} name="upload-file" style="display: none;"
```
Comment (sophie-cluml) [inline src/input/user_input.rs:835]: Is clone necessary here?

Comment (jinjoolee07) [reply src/input/user_input.rs]: Thank you for catching that! I removed the .clone() as it seems unnecessary. Let me know if you'd like me to make any further adjustments!

**Review by sophie-cluml — APPROVED** [review]

File: src/input/config.rs [hunk 3]
```diff
@@ -117,6 +117,7 @@ pub struct NicConfig {
 #[derive(Clone, PartialEq)]
 pub struct FileConfig {
     pub ess: Essential,
+    pub allowed_extensions: Vec<String>,
 }
```
Comment (sehkone) [inline src/input/config.rs:121]: @jinjoolee07 @sophie-cluml 

Is this supposed to include `.` before an extension? I think some documentation on it should be provided.

File: src/input/user_input.rs [hunk 4]
```diff
@@ -807,6 +808,7 @@ where
             }
             Message::ChooseFile(my_index_clone.clone(), result, input_data_clone.clone())
         });
+        let extensions = extensions.join(",");
```
Comment (sehkone) [inline src/input/user_input.rs:811]: @sophie-cluml @jinjoolee07

I think we need to validate each value of `extensions` based on the criteria that has been specified.

Comment (jinjoolee07) [reply src/input/user_input.rs:811]: I was wondering if there’s an existing rule or guideline for validating extensions that I should follow, or would you prefer that I define the criteria? If possible, could you provide more details to help ensure alignment with expectations?

File: src/input/config.rs (hunk 3 above)
Comment (sophie-cluml) [reply src/input/config.rs:121]: @jinjoolee07 I would like to suggest creating an issue for the documentation for this and assigning yourself. I assume the PR that would resolve the issue would include a documentation comment right above this field.

**Comment by sehkone** [comment]
@sophie-cluml, @jinjoolee07

We have three options:

1. Allow extensions starting with a `.` and validate the input to exclude invalid ones.
2. Allow extensions starting without a `.` and validate the input to exclude invalid ones.
3. Allow any extensions and automatically add a `.` if necessary.

In any case, documentation will be required.

By the way, should we validate the input to ensure it does not include spaces and exclude such cases?

**Pull request merged**